6_visualize/src/prep_callouts_fun.R

```diff
@@ -0,0 +1,44 @@
+
+prep_callouts_fun <- function(callouts_cfg, dateTime){
+
+  this_date <- as.POSIXct(dateTime, tz = "UTC") #watch timezones if we ever switch from daily data
+  this_date_callouts <- lapply(callouts_cfg, function(x, this_date) {
+    start <- as.POSIXct(x$dates$start, tz = "UTC")
+    end <- as.POSIXct(x$dates$end, tz = "UTC")
+    if(this_date >= start & this_date <= end) {
+      callouts_to_plot <- x
+    } else {
+      callouts_to_plot <- NULL
+    }
+    return(callouts_to_plot)
+  }, this_date)
+
+  # keep only non-NULL elements
+  this_date_callouts <- this_date_callouts[!unlist(lapply(this_date_callouts, is.null))]
+
```

Comment:
might as well add `rm(callouts_cfg, dateTime)` here to keep the closure (plot_fun + data) smaller.

also here, what about changing the structure a touch so that plot_fun is an empty function if there are no callouts on this date? from this line onward:
```r
  rm(callouts_cfg, dateTime)
  n_callouts <- length(this_date_callouts)
  if(n_callouts > 0) {
    plot_fun <- function(){
      # it is up to the user to parse the text to make sure it doesn't end up outside of the margins
      coord_space <- par()$usr
      for(n in 1:n_callouts) {
        ...
      }
    }
  } else {
    rm(this_date_callouts, n_callouts)
    plot_fun <- function() { NULL }
  }
  return(plot_fun)
```